prime-router/src/main/kotlin/cli/ValidateYAMLCommand.kt

```diff
@@ -0,0 +1,123 @@
+package gov.cdc.prime.router.cli
+
+import com.github.ajalt.clikt.core.BadParameterValue
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.core.CliktError
+import com.github.ajalt.clikt.parameters.options.default
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.types.choice
+import com.github.ajalt.clikt.parameters.types.file
+import com.github.ajalt.mordant.rendering.TextColors.green
+import com.github.ajalt.mordant.rendering.TextColors.red
+import gov.cdc.prime.router.config.validation.ConfigurationType
+import gov.cdc.prime.router.config.validation.ConfigurationValidationFailure
+import gov.cdc.prime.router.config.validation.ConfigurationValidationResult
+import gov.cdc.prime.router.config.validation.ConfigurationValidationService
+import gov.cdc.prime.router.config.validation.ConfigurationValidationServiceImpl
+import gov.cdc.prime.router.config.validation.ConfigurationValidationSuccess
+import org.apache.commons.io.FileUtils
+import java.io.File
+
+class ValidateYAMLCommand : CliktCommand(
+        name = "validate-yaml",
+        help = """
+            A CLI command to validate YAML files' structure and values.
+            
+            Examples:
+            ./gradlew primeCLI --args='validate-yaml --type organizations --file path/to/file'
+            ./gradlew primeCLI --args='validate-yaml --type organizations --dir path/to/directory'
+        """.trimIndent()
+    ) {
+
+    private val typeChoices = ConfigurationType::class
+        .sealedSubclasses
+        .associate { it.simpleName!! to it.objectInstance!! }
+
+    private val type by option(
+        "-t", "--type",
+        help = "Specify the type of YAML file to validate."
+    ).choice(
+        choices = typeChoices,
+        ignoreCase = true
+    ).default(ConfigurationType.Organizations)
+
+    private val file by option(
+        "-f", "--file",
+        help = "Path to a YAML file to validate."
+    ).file(
+        mustBeReadable = true,
+        canBeFile = true,
+        canBeDir = false
+    )
+
+    private val directory by option(
+        "-d", "--dir",
+        help = "Path to a directory containing multiple YAML files to validate."
+    ).file(
+        mustBeReadable = true,
+        canBeFile = false,
+        canBeDir = true
+    )
+
+    val service: ConfigurationValidationService = ConfigurationValidationServiceImpl()
+
+    override fun run() {
+        if (file != null) {
+            handleFile(file!!)
```

Comment:
smart casts don't work with the fancy way Clikt binds arguments to class variables so had to use the forbidden double-bang (`!!`)